quixstreams/sinks/base/sink.py

```diff
@@ -73,8 +102,16 @@ class BatchingSink(BaseSink):
 
     _batches: Dict[Tuple[str, int], SinkBatch]
 
-    def __init__(self):
+    def __init__(
+        self,
+        client_connect_cb: Optional[Callable[[Optional[Exception]], None]] = None,
+    ):
+        """
+        :param client_connect_cb: An optional callback made once a client connection
+            is established. Callback expects an Exception or None as an argument.
+        """
         self._batches = {}
+        super().__init__(client_connect_cb=client_connect_cb)
```

Comment:
Maybe `BatchingSink` doesn't need to know what parent class `__init__` is up to, just make sure it is called with whatever we throw at it.

I think it may safely run before initializing `_baches`.

```suggestion
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self._batches = {}
```

Removes some verbosity. But up to you 😉